mne/io/brainvision/brainvision.py

```diff
@@ -136,12 +138,38 @@ def __init__(self, vhdr_fname, montage=None,
                 offsets = None
                 n_samples = n_samples // (dtype_bytes * (info['nchan'] - 1))
         self.preload = False  # so the event-setting works
+
+        # Get annotations from vmrk file and then use events_from_annotations
+        trig_shift_by_type = _check_trig_shift_by_type(trig_shift_by_type)
+        annots = read_annotations_brainvision(mrk_fname, info['sfreq'])
```

Comment:
it would be great if we can avoid passing `info['sfreq']`. `test.vmrk` has `DataFile=test.eeg` field, but then we should load again the datafile. 
But this is already done in line **118**:
```py
        info, data_filename, fmt, order, mrk_fname, montage, n_samples = \
            _get_vhdr_info(vhdr_fname, eog, misc, scale, montage) 
```

So I'm not sure if we should try to remove sfreq, or actually add sfreq to the rest of `read_annotations_XXX`

Reply:
the sfreq is given in the vhdr file which we don't have. I just pushed a "auto" parameter to read_annotations_brainvision to sfreq is looked in the header file with the same basename.